pkg/kubectl/cmd/logs.go

```diff
@@ -162,18 +172,31 @@ func (o *LogsOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Com
 	o.ClientMapper = resource.ClientMapperFunc(f.ClientForMapping)
 	o.Out = out
 
+	if selector != "" {
+		if logOptions.Follow {
+			return cmdutil.UsageError(cmd, "only one of follow (-f) or selector (-l) is allowed")
```

Comment:
The rules stated in the issue was to disallow follow when using a selector. The concern was that you could be following more than one log which would be more an issue with your cluster and then you would also have to tag every log line.